lib/ansible/modules/remote_management/manageiq/manageiq_provider.py

```diff
@@ -0,0 +1,523 @@
+                    'security_protocol': endpoint.get('security_protocol'),
+                    'certificate_authority': endpoint.get('certificate_authority'),
+                },
+                'authentication': {
+                    'authtype': provider_defaults['authtype'],
+                    'userid': endpoint.get('userid'),
+                    'password': endpoint.get('password'),
+                    'auth_key': endpoint.get('auth_key'),
+                }
+            })
+
+        # build metrics endpoint
+        endpoint = raw_endpoints.get('metrics')
+        if endpoint:
+            endpoints.append({
+                'endpoint': {
+                    'role': endpoint.get('role') or provider_defaults['metrics_role'],
+                    'hostname': endpoint['hostname'],
+                    'port': endpoint['port'],
+                    'verify_ssl': [0, 1][endpoint.get('verify_ssl', True)],
+                    'security_protocol': endpoint.get('security_protocol'),
+                    'certificate_authority': endpoint.get('certificate_authority'),
+                },
+                'authentication': {
+                    'authtype': endpoint.get('role') or provider_defaults['metrics_role'],
+                    'userid': endpoint.get('userid'),
+                    'password': endpoint.get('password'),
+                    'auth_key': endpoint.get('auth_key') or default_auth_key,
+                }
+            })
+
+        return endpoints
+
+    def delete_provider(self, provider):
+        """ Deletes a provider from manageiq.
+
+        Returns:
+            a short message describing the operation executed.
+        """
+        try:
+            url = '%s/providers/%s' % (self.api_url, provider['id'])
+            result = self.client.post(url, action='delete')
+        except Exception as e:
+            self.module.fail_json(msg="failed to delete provider %s: %s" % (provider['name'], str(e)))
+
+        return dict(changed=True, msg=result['message'])
+
+    def edit_provider(self, provider, name, provider_type, endpoints, zone_id, region):
+        """ Edit a user from manageiq.
+
+        Returns:
+            a short message describing the operation executed.
+        """
+        url = '%s/providers/%s' % (self.api_url, provider['id'])
+
+        resource = dict(
+            name=name,
+            zone={'id': zone_id},
+            provider_region=region,
+            connection_configurations=endpoints,
+        )
+
+        # NOTE: we do not check for diff's between requested and current
+        #       provider, we always submit endpoints with password or auth_keys,
+        #       since we can not compare with current password or auth_key,
+        #       every edit request is sent to ManageIQ API without compareing
+        #       it to current state.
+
+        # clean nulls, we do not send nulls to the api
+        resource = delete_nulls(resource)
```

Comment:
:+1: this works around https://github.com/ManageIQ/manageiq/pull/13400 bug, and possibly helps with older API versions that had less fields (as long playbook doesn't specify the new fields).

Subtle: here you omit top-level fields, in create only inside endpoints.
Does this allow you to do partial edit?